examples/vanilla-es6/src/template.js

```diff
@@ -0,0 +1,92 @@
+const htmlEscapes = {
+	'&': '&amp',
+	'<': '&lt',
+	'>': '&gt',
+	'"': '&quot',
+	'\'': '&#x27',
+	'`': '&#x60'
+};
+
+const reUnescapedHtml = /[&<>"'`]/g;
+const reHasUnescapedHtml = new RegExp(reUnescapedHtml.source);
+
+let escape = str => (str && reHasUnescapedHtml.test(str)) ? str.replace(reUnescapedHtml, escapeHtmlChar) : str;
+let escapeHtmlChar = chr => htmlEscapes[chr];
+
+export default class Template {
+	constructor() {
+		this.defaultTemplate = `
+			<li data-id="{{id}}" class="{{completed}}">
+				<div class="view">
+					<input class="toggle" type="checkbox" {{checked}}>
+					<label>{{title}}</label>
+					<button class="destroy"></button>
+				</div>
+			</li>
+		`;
+	}
+
+	/**
+	 * Creates an <li> HTML string and returns it for placement in your app.
+	 *
+	 * NOTE: In real life you should be using a templating engine such as Mustache
+	 * or Handlebars, however, this is a vanilla JS example.
+	 *
+	 * @param {object} data The object containing keys you want to find in the
+	 *                      template to replace.
+	 * @returns {string} HTML String of an <li> element
+	 *
+	 * @example
+	 * view.show({
+		 *	id: 1,
+		 *	title: "Hello World",
+		 *	completed: 0,
+		 * })
+	 */
+	show(data){
+		let i = 0;
+		let view = '';
+		const len = data.length;
+
+		for (i; i < len; i++) {
```

Comment:
Couldn't we map over the data instead and turn it into a list of views instead that we then join together?

E.g. `data.map((d) => ...).join('\n')`